sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -109,6 +109,41 @@ trait StateStore {
   def hasCommitted: Boolean
 }
 
+/** A versioned key-value store which is same as [[StateStore]], but write-protected. */
```

Comment:
For write-protected, we have `WrappedReadOnlyStateStore`. "write-protected" sounds like it can be written, but just being protected. However, `ReadOnlyStateStore` is actually read-only state store. Can we update this comment?

Reply:
read-only would be clearer. Will update.